cg/meta/delivery/delivery.py

```diff
@@ -0,0 +1,170 @@
+        for sample in case.samples:
+            fastq_delivery_files: list[DeliveryFile] = self.get_fastq_delivery_files_by_sample(
+                case=case, sample=sample, force=force
+            )
+            delivery_files.extend(fastq_delivery_files)
+        return delivery_files
+
+    def get_fastq_delivery_files_by_sample(
+        self, case: Case, sample: Sample, force: bool = False
+    ) -> list[DeliveryFile]:
+        """Return a list of fastq files to be delivered for a specific sample."""
+        delivery_files: list[DeliveryFile] = []
+        fastq_tags: list[set[str]] = self.get_analysis_sample_tags_for_workflow(Workflow.FASTQ)
+        if self.is_sample_deliverable(sample=sample, force=force):
+            fastq_files: list[File] = (
+                self.housekeeper_api.get_files_from_latest_version_by_list_of_tags(
+                    bundle_name=sample.internal_id, tags=fastq_tags
+                )
+            )
+            delivery_files: list[DeliveryFile] = self.convert_files_to_delivery_files(
+                files=fastq_files,
+                case=case,
+                source_id=sample.internal_id,
+                destination_id=sample.name,
+            )
+        return delivery_files
+
+    def get_analysis_case_delivery_files(self, case: Case) -> list[DeliveryFile]:
+        """
+        Return a complete list of analysis case files to be delivered and ignore analysis sample
+        files.
+        """
+        case_tags: list[set[str]] = self.get_analysis_case_tags_for_workflow(case.data_analysis)
+        sample_ids: list[set[str]] = [{sample_id} for sample_id in case.sample_ids]
+        case_files: list[File] = self.housekeeper_api.get_files_from_latest_version_by_list_of_tags(
+            bundle_name=case.internal_id, tags=case_tags, excluded_tags=sample_ids
+        )
+        delivery_files: list[DeliveryFile] = self.convert_files_to_delivery_files(
+            files=case_files, case=case, source_id=case.internal_id, destination_id=case.name
+        )
+        return delivery_files
+
+    def get_analysis_sample_delivery_files(self, case: Case) -> list[DeliveryFile]:
+        """Return a complete list of analysis sample files to be delivered."""
+        delivery_files: list[DeliveryFile] = []
+        for sample in case.samples:
+            sample_delivery_files: list[DeliveryFile] = (
+                self.get_analysis_sample_delivery_files_by_sample(case=case, sample=sample)
+            )
+            delivery_files.extend(sample_delivery_files)
+        return delivery_files
+
+    def get_analysis_sample_delivery_files_by_sample(
+        self, case: Case, sample: Sample
+    ) -> list[DeliveryFile]:
+        """Return a list of analysis files to be delivered for a specific sample."""
+        sample_tags: list[set[str]] = self.get_analysis_sample_tags_for_workflow(case.data_analysis)
+        analysis_sample_tags = [tag | {sample.internal_id} for tag in sample_tags]
```

Comment:
I've renamed it and added a type hint. I don't think we need an additional function since it simply performs a union between sets (sample tags and the sample ID).